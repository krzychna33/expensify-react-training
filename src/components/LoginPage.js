import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';

export class LoginPage extends React.Component {
    render(){
        return (
            <div>
                <h1>Expensify app</h1>
                <h4>Please log in.</h4>
                <button onClick={this.props.startLogin}>Login</button>
            </div>
        )
    }
};

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);