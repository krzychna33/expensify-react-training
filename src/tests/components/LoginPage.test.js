import React from 'react';
import {shallow} from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('Should render loginPage', () => {
    const wrapper = shallow(<LoginPage/>);
    expect(wrapper).toMatchSnapshot();
});

test('Should call startLogin', () => {
    const startLoginSpy = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLoginSpy}/>);
    wrapper.find('button').simulate('click')
    expect(startLoginSpy).toHaveBeenCalled();
});