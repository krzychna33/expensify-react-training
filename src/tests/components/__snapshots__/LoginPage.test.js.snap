// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render loginPage 1`] = `
<div>
  <h1>
    Expensify app
  </h1>
  <h4>
    Please log in.
  </h4>
  <button>
    Login
  </button>
</div>
`;
